### PR TITLE
feat: Support python 3.12

### PR DESCRIPTION
**Pull request by gschurck**
Working on python 3.12 support

### CODE REVIEW

File: qiskit_alice_bob_provider/remote/display.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def _new_line_ipython() -> None:
 
 
 def _display_ipython(text: str) -> None:
-    current_label.value = f'[{datetime.now().strftime("%H:%M:%S")}] {text}'
+    current_label.value = f"""[{datetime.now().strftime('%H:%M:%S')}] {text}"""
```
Comment (sbrieuc) [inline qiskit_alice_bob_provider/remote/display.py:36]: I don't understand how this change is related to Python 3.12 support? Is it purely a style change? If yes I don't see the point as we use single quotes everywhere in the code.

File: qiskit_alice_bob_provider/remote/display.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def _display_std(text: str) -> None:
     # We use a padding of 80 characters to be sure everything is flushed out
     # when we do the next call to this function.
     print(
-        f'[{datetime.now().strftime("%H:%M:%S")}] {text:<80}',
+        f"""[{datetime.now().strftime('%H:%M:%S')}] {text:<80}""",
```
Comment (sbrieuc) [inline qiskit_alice_bob_provider/remote/display.py:47]: Same remark, this change does not seem necessary for Python 3.12 support.

File: qiskit_alice_bob_provider/remote/job.py [hunk 3]
```diff
@@ -210,14 +210,16 @@ def _monitor_state(
             ):
                 success_msg += (
                     ' Time spent executing on QPU: '
-                    f'{_format_nanoseconds(metrics["qpu_duration_ns"])}.'
+                    f"""{_format_nanoseconds(metrics['qpu_duration_ns'])}."""
```
Comment (sbrieuc) [inline qiskit_alice_bob_provider/remote/job.py:213]: Same remark, this change does not seem necessary for Python 3.12 support.

File: qiskit_alice_bob_provider/remote/job.py [hunk 4]
```diff
@@ -210,14 +210,16 @@ def _monitor_state(
...
                 )
             elif 'simulation_duration_ns' in metrics and isinstance(
                 metrics['simulation_duration_ns'], int
             ):
                 success_msg += (
                     ' Time spent simulating: '
-                    f'{_format_nanoseconds(metrics["simulation_duration_ns"])}'
+                    f"""{_format_nanoseconds(
+                        metrics['simulation_duration_ns']
+                    )}"""
```
Comment (sbrieuc) [inline qiskit_alice_bob_provider/remote/job.py:222]: Same remark, this change does not seem necessary for Python 3.12 support.

**Review by sbrieuc — CHANGES_REQUESTED** [review]

File: qiskit_alice_bob_provider/remote/display.py (hunk 1 above)
Comment (gschurck) [reply qiskit_alice_bob_provider/remote/display.py:36]: I updated to `pylint` v3 and `flakes8-quotes` 3.4 to make them compatible with python 3.12. I remember I had some trouble with specific rules and I changed the quotes, but I can't reproduce it anymore. I think I just misunderstood their behavior, these changes are indeed not needed.

File: pyproject.toml [hunk 5]
```diff
@@ -21,19 +21,18 @@ classifiers=[
     "Operating System :: MacOS",
     "Operating System :: POSIX :: Linux",
     "Programming Language :: Python :: 3 :: Only",
-    "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
```
Comment (alxthm) [inline pyproject.toml:26]: should we add a line with python 3.12 ?

File: pyproject.toml [hunk 6]
```diff
@@ -21,19 +21,18 @@ classifiers=[
     "Operating System :: MacOS",
     "Operating System :: POSIX :: Linux",
     "Programming Language :: Python :: 3 :: Only",
-    "Programming Language :: Python :: 3.8",
```
Comment (alxthm) [inline pyproject.toml:24]: we are removing python 3.8 here, but not in `requires-python` below, is that on purpose ?

File: pyproject.toml (hunk 5 above)
Comment (gschurck) [reply pyproject.toml:26]: Yes ! Forgot it

**Review by alxthm — APPROVED** [review]

**Review by sbrieuc — APPROVED** [review]

**Pull request merged**